File: shuttle-api/src/ShuttleApi.Domain/Trips/TripPostReport.cs
using ShuttleApi.Domain.Common;

namespace ShuttleApi.Domain.Trips;

public sealed class TripPostReport : Entity<Guid>
{
    public Guid TripId { get; private set; }
    public int OdometerStart { get; private set; }
    public int OdometerEnd { get; private set; }
    public int DistanceKm => OdometerEnd - OdometerStart;
    public decimal? FuelAddedLitres { get; private set; }
    public decimal? FuelCostDollars { get; private set; }
    public bool HasIncident { get; private set; }
    public IncidentType? IncidentType { get; private set; }
    public string? IncidentDescription { get; private set; }
    public string? AdditionalNotes { get; private set; }
    public DateTime SubmittedAt { get; private set; }
    public bool IsReadyToInvoice { get; private set; }

    // Post-trip vehicle checklist (Section 9 of physical manifest)
    public bool ExteriorNoNewDamage { get; private set; }
    public bool InteriorCleanedAndChecked { get; private set; }
    public bool PassengersDisembarkedSafely { get; private set; }
    public bool AllCargoDeliveredAndAccounted { get; private set; }
    public bool VehicleSecuredAndPluggedIn { get; private set; }
    public bool KeysReturnedAndSecured { get; private set; }

    private TripPostReport() { }

    public static TripPostReport Create(
        Guid tripId,
        int odometerStart,
        int odometerEnd,
        decimal? fuelAddedLitres,
        decimal? fuelCostDollars,
        bool hasIncident,
        IncidentType? incidentType,
        string? incidentDescription,
        string? additionalNotes,
        bool isReadyToInvoice,
        bool exteriorNoNewDamage = false,
        bool interiorCleanedAndChecked = false,
        bool passengersDisembarkedSafely = false,
        bool allCargoDeliveredAndAccounted = false,
        bool vehicleSecuredAndPluggedIn = false,
        bool keysReturnedAndSecured = false)
    {
        return new TripPostReport
        {
            Id = Guid.NewGuid(),
            TripId = tripId,
            OdometerStart = odometerStart,
            OdometerEnd = odometerEnd,
            FuelAddedLitres = fuelAddedLitres,
            FuelCostDollars = fuelCostDollars,
            HasIncident = hasIncident,
            IncidentType = incidentType,
            IncidentDescription = incidentDescription,
            AdditionalNotes = additionalNotes,
            SubmittedAt = DateTime.UtcNow,
            IsReadyToInvoice = isReadyToInvoice,
            ExteriorNoNewDamage = exteriorNoNewDamage,
            InteriorCleanedAndChecked = interiorCleanedAndChecked,
            PassengersDisembarkedSafely = passengersDisembarkedSafely,
            AllCargoDeliveredAndAccounted = allCargoDeliveredAndAccounted,
            VehicleSecuredAndPluggedIn = vehicleSecuredAndPluggedIn,
            KeysReturnedAndSecured = keysReturnedAndSecured
        };
    }
}

File: shuttle_tablet/lib/features/trips/domain/entities/trip_post_report.dart
import 'package:equatable/equatable.dart';

enum IncidentType { delay, passengerNoShow, vehicleIssue, cargoDamage, accident }

class TripPostReport extends Equatable {
  final String id;
  final String tripId;
  final int odometerStart;
  final int odometerEnd;
  final int distanceKm;
  final double? fuelAddedLitres;
  final double? fuelCostDollars;
  final bool hasIncident;
  final IncidentType? incidentType;
  final String? incidentDescription;
  final String? additionalNotes;
  final DateTime submittedAt;
  final bool isReadyToInvoice;
  final bool exteriorNoNewDamage;
  final bool interiorCleanedAndChecked;
  final bool passengersDisembarkedSafely;
  final bool allCargoDeliveredAndAccounted;
  final bool vehicleSecuredAndPluggedIn;
  final bool keysReturnedAndSecured;

  const TripPostReport({
    required this.id,
    required this.tripId,
    required this.odometerStart,
    required this.odometerEnd,
    required this.distanceKm,
    this.fuelAddedLitres,
    this.fuelCostDollars,
    required this.hasIncident,
    this.incidentType,
    this.incidentDescription,
    this.additionalNotes,
    required this.submittedAt,
    required this.isReadyToInvoice,
    this.exteriorNoNewDamage = false,
    this.interiorCleanedAndChecked = false,
    this.passengersDisembarkedSafely = false,
    this.allCargoDeliveredAndAccounted = false,
    this.vehicleSecuredAndPluggedIn = false,
    this.keysReturnedAndSecured = false,
  });

  @override
  List<Object?> get props => [
        id,
        tripId,
        odometerStart,
        odometerEnd,
        distanceKm,
        fuelAddedLitres,
        fuelCostDollars,
        hasIncident,
        incidentType,
        incidentDescription,
        additionalNotes,
        submittedAt,
        isReadyToInvoice,
        exteriorNoNewDamage,
        interiorCleanedAndChecked,
        passengersDisembarkedSafely,
        allCargoDeliveredAndAccounted,
        vehicleSecuredAndPluggedIn,
        keysReturnedAndSecured,
      ];
}
